packages/mdc-auto-init/index.ts

```diff
@@ -41,23 +43,25 @@ function _emit(evtType, evtData, shouldBubble = false) {
 }
 
 /**
- * Auto-initializes all mdc components on a page.
+ * Auto-initializes all MDC components on a page.
  */
-export default function mdcAutoInit(root = document, warn = CONSOLE_WARN) {
+function mdcAutoInit(root = document, warn = CONSOLE_WARN) {
   const components = [];
-  const nodes = root.querySelectorAll('[data-mdc-auto-init]');
-  for (let i = 0, node; (node = nodes[i]); i++) {
+  const nodes: HTMLElement[] = [].slice.call(root.querySelectorAll<HTMLElement>('[data-mdc-auto-init]'));
+
+  for (const node of nodes) {
     const ctorName = node.dataset.mdcAutoInit;
     if (!ctorName) {
       throw new Error('(mdc-auto-init) Constructor name must be given.');
     }
 
-    const Ctor = registry[ctorName];
+    const Ctor = registry[ctorName]; // tslint:disable-line:variable-name
     if (typeof Ctor !== 'function') {
       throw new Error(
         `(mdc-auto-init) Could not find constructor in registry for ${ctorName}`);
     }
 
+    // @ts-ignore
```

Comment:
if we have a ts-ignore, we should also have the reason why we need to ignore it - please comment.